app/javascript/components/util/DateInput.js

```diff
@@ -23,15 +23,11 @@ class DateInput extends React.Component {
    */
   handleDateChange = date => {
     const momentDate = date && (this.props.showTime ? formatTimestamp(moment(date)) : moment(date).format('YYYY-MM-DD'));
+    this.props.onChange(this.props.showTime ? date : momentDate);
     this.datePickerRef.current.setOpen(false);
     this.setState({ currentDate: momentDate });
     if (this.dateIsValidAndNotEmpty(date)) {
-      this.setState({ lastValidDate: date });
-      // only call onChange if date is valid and present when date is required
-      this.props.onChange(date);
-    } else if (!this.props.required) {
-      // always call onChange when date is not required
-      this.props.onChange(date);
+      this.setState({ lastValidDate: this.props.showTime ? date : momentDate });
```

Comment:
@timwongj is better suited to look at this than I, but I know he made the changes this reverts to prevent a bug that lets you set a date in the future. See the recording below for an example. Even though it appears the frontend reverts a future date to the current time, the state is still updated to the future time

https://user-images.githubusercontent.com/52747882/140823197-56161a37-5da8-4230-a9f1-1d60280b046b.mov

.

Reply:
Ah I see, thank you. Reverted that part https://github.com/SaraAlert/SaraAlert/pull/1156/commits/09afc4e595773c28712bdc1087a1625aed624b33